vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -0,0 +1,137 @@
+    )
+
+
+def match_any_regex(string, regexes):
+    for regex in regexes:
+        match = regex.match(string)
+        if match:
+            return True
+    return False
+
+
+def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
+    resource_type = MOR_TYPE_AS_STRING[type(mor)]
+
+    for f in resource_filters:
+        if f[0] == resource_type:
+            break
+    else:
+        return False
+
+    name_filter = resource_filters.get((resource_type, 'name'))
+    inventory_path_filter = resource_filters.get((resource_type, 'inventory_path'))
+    hostname_filter = resource_filters.get((resource_type, 'hostname'))
+    guest_hostname_filter = resource_filters.get((resource_type, 'guest_hostname'))
+
+    if name_filter:
+        mor_name = infrastructure_data.get(mor).get("name", "")
+        if match_any_regex(mor_name, name_filter):
+            return False
+    if inventory_path_filter:
+        path = make_inventory_path(mor, infrastructure_data)
+        if match_any_regex(path, inventory_path_filter):
+            return False
+
+    if hostname_filter and isinstance(mor, vim.VirtualMachine):
+        host = infrastructure_data.get(mor).get("runtime.host")
+        hostname = infrastructure_data.get(host, {}).get("name", "")
+        if match_any_regex(hostname, hostname_filter):
+            return False
+    if guest_hostname_filter and isinstance(mor, vim.VirtualMachine):
+        guest_hostname = infrastructure_data.get(mor, {}).get("guest.hostName", "")
+        if match_any_regex(guest_hostname, guest_hostname_filter):
+            return False
+
+    return True
+
+
+def is_metric_excluded_by_filters(metric_name, mor_type, metric_filters):
+    filters = metric_filters.get(MOR_TYPE_AS_STRING[mor_type])
+    if not filters:
+        # No filters means collect everything
+        return False
+    if match_any_regex(metric_name, filters):
+        return False
+
+    return True
+
+
+def make_inventory_path(mor, infrastructure_data):
+    mor_name = infrastructure_data.get(mor).get('name', '')
+    mor_parent = infrastructure_data.get(mor).get('parent')
+    if mor_parent:
+        return make_inventory_path(mor_parent, infrastructure_data) + '/' + mor_name
+    return ''
+
+
+def get_parent_tags_recursively(mor, infrastructure_data):
+    """Go up the resources hierarchy from the given mor. Note that a host running a VM is not considered to be a
+    parent of that VM.
+
+    rootFolder(vim.Folder):
+      - vm(vim.Folder):
+          VM1-1
+          VM1-2
+      - host(vim.Folder):
+          HOST1
+          HOST2
+
+    """
+    mor_props = infrastructure_data.get(mor)
+    parent = mor_props.get('parent')
+    if parent:
+        tags = []
+        parent_props = infrastructure_data.get(parent, {})
+        parent_name = ensure_unicode(parent_props.get('name', 'unknown'))
+        if isinstance(parent, vim.HostSystem):
+            tags.append(u'vsphere_host:{}'.format(parent_name))
```

Comment:
Any reason to use `u` unicode strings instead of native type?

Same comment for all other cases.

Reply:
The "name" property can contain unicode character, so in order to parse them correctly they are passed inside `ensure_unicode` method to reflect correctly the non-standard character.

And in order to do formatting you cannot mix non-unicode with unicode if you don't want to see an exception.

Reply:
Done